apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -190,14 +198,32 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
 
       // We don't use JsonFile/jju here because shrinkwrap.json is a special NPM file format
       // and typically very large, so we want to load it the same way that NPM does.
-      const parsedData: IPnpmShrinkwrapYaml = yaml.safeLoad(FileSystem.readFile(shrinkwrapYamlFilename).toString());
+      const shrinkwrapContent: string = FileSystem.readFile(shrinkwrapYamlFilename).toString();
+      const parsedData: IPnpmShrinkwrapYaml = yaml.safeLoad(shrinkwrapContent);
 
-      return new PnpmShrinkwrapFile(parsedData, shrinkwrapYamlFilename);
+      // Grab the shrinkwrap hash out of the comment where we store it.
+      const hashIndex: number = shrinkwrapContent.lastIndexOf(PnpmShrinkwrapFile._shrinkwrapHashPrefix);
+      const shrinkwrapHash: string | undefined = hashIndex >= 0
```

Comment:
```
const hashIndex: number = shrinkwrapContent.lastIndexOf(PnpmShrinkwrapFile._shrinkwrapHashPrefix);
```

This feels uncool to me.  For example, it could incorrectly match something like this:

```yaml
lockfileVersion: 5
packages:
  /@babel/code-frame/7.8.3:
    dependencies:
      '@babel/highlight': 7.8.3
    dev: false
    resolution:
      integrity: '# shrinkwrap hash: '
  /@babel/highlight/7.8.3:
    dependencies:
      chalk: 2.4.2
      esutils: 2.0.3
      js-tokens: 4.0.0
    dev: false
```

Such an input is statistically unlikely to happen, but it makes me uncomfortable to use wrong algorithms that "probably" will work.  Junior engineers learn primarily from reading existing code, so it's useful for our code to demonstrate correct algorithms when not strictly required.

Here's how to solve it with a RegExp:
```js
const hashYamlCommentRegExp = /\n\s*# shrinkwrap hash:\s*\n\s*(\S+)\s*$/;
const match = yamlText.match(hashYamlCommentRegExp);
if (match) {
  console.log('The hash is ' + match[1]);
}
```